File: nvxapp.server.service/ClientServer-Service/GestionePresenze/Par_CausaliService/Models/Par_CausaliModel.cs
using nvxapp.server.service.ClientServer_Service.ModelsBase;
using System.Collections.Generic;

namespace nvxapp.server.service.ClientServer_Service.GestionePresenze.Par_CausaliService.Models
{
    public class Par_CausaliModel
    {
        public int Id { get; set; }
        public int IdAz_Anagrafica { get; set; }
        public string? Codice { get; set; }
        public string? Descrizione { get; set; }
    }

    public class Par_CausaliInModel
    {

    }

    public class Par_CausaliOutModel : ModelResult
    {
        public List<Par_CausaliModel> Par_Causali { get; set; } = new List<Par_CausaliModel>();
        public Par_CausaliOutModel()
        {

        }
    }

    public class Par_CausaliGetInModel
    {
        public int Id { get; set; } = 0;
    }
    public class Par_CausaliGetOutModel : ModelResult
    {
        public Par_CausaliModel Par_Causale { get; set; } = new Par_CausaliModel();
    }
    public class Par_CausaliPutInModel : ModelResult
    {
        public Par_CausaliModel Par_Causale { get; set; } = new Par_CausaliModel();
    }
    public class Par_CausaliPutOutModel : ModelResult
    {
        public Par_CausaliModel Par_Causale { get; set; } = new Par_CausaliModel();
    }

    public class Par_Causali_DeleteInModel
    {
        public int Id { get; set; }
    }
    public class Par_Causali_DeleteOutModel : ModelResult { }
}

File: nvxapp.client/src/nvx/ClientServer-Service/GestionePresenze/Par_Causali/Models/par-causali-model.ts
import { ModelResult } from "../../../ModelsBase/model-result";

export class Par_CausaliModel {
    public id!: number;
    public idAz_Anagrafica!: number;
    public descrizione!: string;
    public codice!: string;
}

export class Par_CausaliInModel {
}

export class Par_CausaliOutModel extends ModelResult {
    public par_Causali: Par_CausaliModel[];
}

export class Par_CausaliGetInModel {
    public id: number;
}

export class Par_CausaliGetOutModel extends ModelResult {
    public par_Causale: Par_CausaliModel;
}

export class Par_CausaliPutInModel {
    public par_Causale: Par_CausaliModel;
}

export class Par_CausaliPutOutModel extends ModelResult {
    public par_Causale: Par_CausaliModel;
}

export class Par_Causali_DeleteInModel {
  id: number;
}
export class Par_Causali_DeleteOutModel extends ModelResult {
}
